hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/NameNode.java

```diff
@@ -1899,6 +1899,9 @@ synchronized void transitionToStandby() throws IOException {
   synchronized void transitionToObserver() throws IOException {
     String operationName = "transitionToObserver";
     namesystem.checkSuperuserPrivilege(operationName);
+    if (notBecomeActiveInSafemode && isInSafeMode()) {
```

Comment:
Can we use a common flag?  `notBecomeActiveInSafemode` is used to control Active.

Reply:
Hi @ZanderXu , @xkrogen mentioned earlier that this parameter is used for guard and document this new behavior. What do you think of this?

Reply:
Updated the document.